pinot-core/src/main/java/org/apache/pinot/core/common/DataBlockCache.java

```diff
@@ -255,9 +255,10 @@ public String[] getStringValuesForSVColumn(String column) {
    * @param column Column name
    * @param evaluator transform evaluator
    * @param buffer values to fill
+   * @param parseExactBigDecimal parse exact BigDecimal values
    */
-  public void fillValues(String column, TransformEvaluator evaluator, String[] buffer) {
-    _dataFetcher.fetchStringValues(column, evaluator, _docIds, _length, buffer);
+  public void fillValues(String column, TransformEvaluator evaluator, String[] buffer, boolean parseExactBigDecimal) {
```

Comment:
We don't need to parse big decimal in the physical data access layer. It should be read as `byte[]`. It is not feasible to add physical data access for every individual logical types.

Reply:
@Jackie-Jiang Many transformations (e.g. `JsonExtractScalarTransformFunction`, `PushDownTransformFunction`) don't override `transformToBytesValuesSV` from `BaseTransformFunction` while it overrides `transformTo[INT|LONG|FLOAT|DOUBLE]ValuesSV`. Also, `ProjectionBlock` and `DataBlockCache` don't implement a method that takes TransformEvaluator: `fillValues(ProjectionBlock, TransformEvaluator, byte[][])`.

In case of `JsonExtractScalarTransformFunction`, we need to read BigDecimals as strings rather than bytes. I will simplify the logic and abstract the details of BigDecimal away.